ethportal-api/src/types/content_key/state.rs

```diff
@@ -143,224 +157,117 @@ impl fmt::Display for StateContentKey {
-        let key = StateContentKey::AccountTrieProof(proof);
-        let encoded: Vec<u8> = key.clone().into();
+        let key = StateContentKey::ContractBytecode(ContractBytecodeKey {
+            address: Address::from(ADDRESS),
+            code_hash: H256::from_str(CODE_HASH).unwrap(),
+        });
 
-        assert_eq!(hex_decode(expected_content_key).unwrap(), encoded);
-        assert_eq!(expected_content_id, key.content_id());
+        assert_encode_decode(&key);
+        assert_content_key_and_id(&key, expected_content_key, expected_content_id);
     }
 
     #[test]
-    fn contract_storage_trie_proof() {
-        let expected_content_key = "0x03829bd824b016326a401d083b33d092293333a830c8a6030000000000000000000000000000000000000000000000000000000000d1c390624d3bd4e409a61a858e5dcc5517729a9170d014a6c96530d64dd8621d";
-        let expected_content_id: [u8; 32] = [
-            0xb1, 0xc8, 0x99, 0x84, 0x80, 0x3c, 0xeb, 0xd3, 0x25, 0x30, 0x3b, 0xa0, 0x35, 0xf9,
-            0xc4, 0xca, 0x0d, 0x0d, 0x91, 0xb2, 0xcb, 0xfe, 0xf8, 0x4d, 0x45, 0x5e, 0x7a, 0x84,
-            0x7a, 0xde, 0x1f, 0x08,
-        ];
-
-        let slot = U256::from(239304u128);
-        let proof = ContractStorageTrieProof {
-            address: ADDRESS.to_vec().into(),
-            slot,
-            state_root: STATE_ROOT,
-        };
-
-        let key = StateContentKey::ContractStorageTrieProof(proof);
-        let encoded: Vec<u8> = key.clone().into();
-
-        assert_eq!(hex_decode(expected_content_key).unwrap(), encoded);
-        assert_eq!(expected_content_id, key.content_id());
+    fn decode_empty_key_should_fail() {
+        assert_eq!(
+            StateContentKey::try_from(vec![]).unwrap_err().to_string(),
+            "Unable to decode key SSZ bytes 0x due to InvalidLengthPrefix { len: 0, expected: 1 }",
+        );
     }
 
     #[test]
-    fn contract_bytecode() {
-        let expected_content_key = "0x04829bd824b016326a401d083b33d092293333a830d1c390624d3bd4e409a61a858e5dcc5517729a9170d014a6c96530d64dd8621d";
-        let expected_content_id: [u8; 32] = [
-            0x14, 0x6f, 0xb9, 0x37, 0xaf, 0xe4, 0x2b, 0xcf, 0x11, 0xd2, 0x5a, 0xd5, 0x7d, 0x67,
-            0x73, 0x4b, 0x9a, 0x71, 0x38, 0x67, 0x7d, 0x59, 0xee, 0xec, 0x3f, 0x40, 0x29, 0x08,
-            0xf5, 0x4d, 0xaf, 0xb0,
-        ];
-
-        let bytecode = ContractBytecode {
-            address: ADDRESS.to_vec().into(),
-            code_hash: CODE_HASH,
-        };
+    fn decode_key_with_invalid_selector_should_fail() {
+        let invalid_selector_content_key = "0x0024000000c5d2460186f7233c927e7db2dcc703c0e500b653ca82273b7bfad8045d85a4700005000000";
+        assert_eq!(
+            StateContentKey::try_from(hex_decode(invalid_selector_content_key).unwrap())
+                .unwrap_err()
+                .to_string(),
+            "Unable to decode key SSZ bytes 0x0024000000c5d2460186f7233c927e7db2dcc703c0e500b653ca82273b7bfad8045d85a4700005000000 due to UnionSelectorInvalid(0)",
```

Comment:
I would probably use `format!()`/string interpolation here so that you can re-use the `invalid_selector_content_key` variable, without having to duplicate the actual value.

Reply:
Oops. Thanks. Done.